tests/helpers/test_template.py

```diff
@@ -475,6 +477,171 @@ def test_isnumber(hass: HomeAssistant, value, expected) -> None:
     )
 
 
+@pytest.mark.parametrize(
+    ("value", "expected"),
+    [
+        ([1, 2], True),
+        ({1, 2}, False),
+        ({"a": 1, "b": 2}, False),
+        (ReadOnlyDict({"a": 1, "b": 2}), False),
+        (MappingProxyType({"a": 1, "b": 2}), False),
+        ("abc", False),
+        (b"abc", False),
+        ((1, 2), False),
+        (datetime(2024, 1, 1, 0, 0, 0), False),
+    ],
+)
+def test_islist(hass: HomeAssistant, value, expected) -> None:
```

Comment:
```suggestion
def test_islist(hass: HomeAssistant, value: Any, expected: bool) -> None:
```

Reply:
Do we actually type these? From other tests I saw that only things provided by HA are typed. I am OK with typing them just want to make sure we aren't breaking standard

Reply:
In my opinion, it is always better to provide more type hints if you can/know the types. With the type hints the IDE and all our checking tools (mypy, ruff, pylint, ...) can give you hints if you are using a variable wrong.